File: app/models/period_department_item.rb
class PeriodDepartmentItem
  include Mongoid::Document
  include Mongoid::Timestamps

  embedded_in :period_department

  field :name, type: String
  field :price, type: BigDecimal
  field :quantity, type: Integer

  field :total_copies, type: Integer
  field :total_deductions, type: Integer
  field :amount, type: BigDecimal

  field :days, type: String

  field :days_off, type: Array
end

File: src/app/models/period-department-item.ts
import { MongoidId } from './mongoid-id';

export interface PeriodDepartmentItemParams {
  price: number;
  name: string;
  days: string;
  quantity: number;
  total_copies: number;
  total_deductions: number;
  days_off?: { date: number; amount: number }[];
}

export interface PeriodDepartmentItem
  extends MongoidId,
    PeriodDepartmentItemParams {}
